aten/src/ATen/native/Pow.h

```diff
@@ -9,6 +9,48 @@ struct TensorIterator;
 
 namespace native {
 
+#if defined(__CUDACC__) || defined(__HIPCC__)
+#define HOST_DEVICE __host__ __device__
+#else
+#define HOST_DEVICE
+#endif
+
+template <class T,
+  typename std::enable_if<std::is_integral<T>::value, T>::type* = nullptr>
+static inline HOST_DEVICE __ubsan_ignore_signed_int_overflow__ T powi_impl(T a, T b) {
+  T result = 1;
+  while (b) {
+    if (b & 1) {
+       result *= a;
+    }
+    b /= 2;
+    a *= a;
+  }
+  return result;
+}
+
+template <class T,
+  typename std::enable_if<std::is_integral<T>::value && !std::is_signed<T>::value, T>::type* = nullptr>
+static inline HOST_DEVICE T powi(T a, T b) {
+  return powi_impl(a, b);
+}
+
+template <class T,
+  typename std::enable_if<std::is_integral<T>::value && std::is_signed<T>::value, T>::type* = nullptr>
+static inline HOST_DEVICE T powi(T a, T b) {
+  if ( b < 0 ) {
```

Comment:
In NumPy, attempting to raise an integer array to a negative power will throw an error:

```
a = np.array((2))
a ** -1
: ValueError: Integers to negative integer powers are not allowed.
```

Reply:
we could change it, though it would be an additional BC-breaking change. On master `torch.tensor(1) ** torch.tensor(-1)` returns `tensor(1)`. I think if we decide to change that too it could go in a separate PR.

Reply:
Separating them seems OK. Can we put a comment describing the desired behavior in this PR, then?

Reply:
added a comment